exercises/t_jubb/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,270 @@
+		perimeter = 0.0;
+	};
+	Rectangle(double height, double width) {
+		type = "Rectangle";
+		sides = 4;
+		area = height * width;
+		perimeter = 2 * height + 2 * width;
+	};
+
+private:
+	double height, width;
+
+};
+
+/*
+Triangle Class
+*/
+class IsoscelesTriangle : public Shape {
+
+public:
+	IsoscelesTriangle() {
+		type = "Triangle";
+		height = 0.0;
+		base = 0.0;
+		sides = 3;
+		area = 0.0;
+		perimeter = 0.0;
+	};
+	IsoscelesTriangle(double base, double height) {
+		type = "Triangle";
+		sides = 3;
+		area = 0.5 * base * height;
+		perimeter = base + 2 * sqrt(pow(height, 2) + 0.25 * pow(base, 2));
+	};
+
+private:
+	double base, height;
+};
+
+/*
+Circlee Class
+*/
+class Circle : public Shape {
+
+public:
+	Circle() {
+		type = "Circle";
+		radius = 0.0;
+		sides = 1;
+		area = 0.0;
+		perimeter = 0.0;
+	};
+	Circle(double radius) {
+		type = "Circle";
+		sides = 1;
+		area = M_PI * pow(radius, 2);
+		perimeter = M_PI * 2 * radius;
+	};
+
+private:
+	double radius;
+};
+
+/*
+Comparisons between shapes for area, perimeter
+*/
+bool compare_area(Shape& shape1, Shape& shape2) {
+	return shape1.get_area() < shape2.get_area();
+};
+bool compare_perimeter(Shape& shape1, Shape& shape2) {
+	return shape1.get_perimeter() < shape2.get_perimeter();
+};
+
+class ShapeSorter : public Shape {
+
+public:
+	ShapeSorter() { };
+	ShapeSorter(std::vector<Shape> vec) { shape_vector = vec; };
+	void print_type(std::string type);
+	void print_sides(int sides);
+	void print_by_area(bool ascending);
+	void print_by_perimeter(bool ascending);
+
+private:
+	std::vector<Shape> shape_vector;
+
+};
+
+/*
+Prints shapes in shape_vector with a given type.
+*/
+void ShapeSorter::print_type(std::string type) {
+	for (auto i = shape_vector.begin(); i != shape_vector.end(); i++) {
+		if ((*i).get_type() == type) {
+			(*i).print_shape();
+		};
+	};
+};
+
+/*
+Prints shapes in shape_vector with a given number of sides.
+*/
+void ShapeSorter::print_sides(int sides) {
+	for (auto i = shape_vector.begin(); i != shape_vector.end(); i++) {
+		if ((*i).get_sides() == sides) {
+			(*i).print_shape();
+		};
+	};
+};
+
+/*
+
+Sort the vector of Shapes by their area and print out the contents.
+
+@param ascending : Wether to sort from lowest to highst
+
+*/
+void ShapeSorter::print_by_area(bool ascending) {
+
+	sort(shape_vector.begin(), shape_vector.end(),compare_area);
+
+	if (ascending) {
+		for (auto i = shape_vector.begin(); i != shape_vector.end(); i++) {
+			(*i).print_shape();
+		};
+	}
+	else {
+		for (auto i = shape_vector.rbegin(); i != shape_vector.rend(); i++) {
+			(*i).print_shape();
+		};
+
+	}
+	
+};
+
+/*
+
+Sort the vector of Shapes by their perimeter and print out the contents.
+
+@param ascending : Wether to sort from lowest to highst
+
+*/
+void ShapeSorter::print_by_perimeter(bool ascending) {
+
+	sort(shape_vector.begin(), shape_vector.end(), compare_perimeter);
+
+	if (ascending) {
+		for (auto i = shape_vector.begin(); i != shape_vector.end(); i++) {
```

Comment:
I would consider having a private member function to do the printing that can be called by the others.